core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -44,6 +54,78 @@ static GtfsInput createFromPath(String path) throws IOException {
         return new GtfsZipFileInput(file);
     }
 
+    /**
+     * Creates an specific GtfsInput to read data from the given URL.
+     *
+     * @param sourceUrl
+     * @param targetPathAsString
+     * @return
+     * @throws IOException
+     */
+    static GtfsInput createFromUrl(URL sourceUrl, String targetPathAsString) throws IOException {
+        InputStream inputStream;
+
+        try {
+            final HttpURLConnection httpConnection = (HttpURLConnection) sourceUrl.openConnection();
```

Comment:
How about using a higher-level Apache library for HTTP?

It has `org.apache.http.client.methods.HttpGet` class that we can use here. The default redirect strategy handles redirects for GET and HEAD, so that is what we need.

Reply:
I see that you have chosen `org.apache.hc.client5.http.classic.methods.HttpGet` instead but it is not available at Google internally, unfortunately. Please pay attention to the packages that I suggest because we have to stick to those libraries and versions that Google has internally. Otherwise the Java validator cannot run at Google.